python/rmm/_lib/cuda_stream_view.pyx

```diff
@@ -12,21 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
+
 from libc.stdint cimport uintptr_t
 
 
+cpdef enum cudaStream:
+    cudaStreamDefault = 0
+    cudaStreamLegacy = 1
+    cudaStreamPerThread = 2
+
+
 cdef class CudaStreamView:
 
-    def __cinit__(self, uintptr_t stream=0):
+    def __cinit__(self, cudaStream stream=cudaStreamDefault):
         """Construct a view of the specified CUDA stream
 
         Parameters
         ----------
         stream : uintptr_t, optional
             CUDA stream to wrap, default 0
         """
-        if (stream == 0):
-            self.c_obj.reset(new cuda_stream_view())
+        if (stream == cudaStreamDefault):
+            if int(os.environ.get("RMM_PER_THREAD_DEFAULT_STREAM", "0")) != 0:
+                self.c_obj.reset(new cuda_stream_view(<cudaStream_t>cudaStreamPerThread))
+            else:
+                self.c_obj.reset(new cuda_stream_view())
```

Comment:
Would still suggest having the environment variable handled globally instead of locally. That allows this to get set once. Also avoids changes in the environment variable affecting a program part way through in unexpected ways.

Reply:
I'll do that, but it seems like you're suggesting we cache the value of the env var at some point, is that correct? Do you have an idea/place in mind on how to do it?